### PR TITLE
fix(select): remove selectedOptions from controlled select state

### PR DESCRIPTION
**Pull request by schnerd**
Fix for #312 

Since `selectedOptions` is the value of the form control, it should _not_ be part of the state. Since `Select` is intended to be a controlled component, it should always reflect the value passed in through props.

### CODE REVIEW

File: src/select/__tests__/select.test.js [hunk 1]
```diff
@@ -139,74 +125,18 @@ describe('Stateless select', function() {
...
-          };
-          const event =
-            type === STATE_CHANGE_TYPE.unselect
-              ? [e, undefined, true]
-              : [e, selectedOption];
-          wrapper.instance().onChange(...event);
-          expect(allProps.onChange).toHaveBeenCalledWith(
-            e,
-            Object.assign({}, expectedResult, {type}),
-          );
```
Comment (schnerd) [inline src/select/__tests__/select.test.js]: This was testing the internal implementation – we should just be testing the public interface of the component.

Comment (blackswanny) [reply src/select/__tests__/select.test.js]: if this is internal , there has to be the way to test it. For instance to emulate `'click` event here. We shoulnd't just remove tests

File: src/select/__tests__/select.test.js [hunk 2]
```diff
@@ -406,48 +372,31 @@ describe('Stateless select', function() {
           {isDropDownOpen: mode === TYPE.select},
         ],
         [KEY_STRINGS.Escape, 'close dropdown', {isDropDownOpen: false}],
-        [
```
Comment (blackswanny) [inline src/select/__tests__/select.test.js]: why is this test moved to another place. We re-used more code in here and it was still readable

File: src/select/__tests__/select.test.js (hunk 1 above)
Comment (schnerd) [reply src/select/__tests__/select.test.js]: yeah I guess I assumed this behavior was covered by another test that actually simulated click events, but I see now that it was not. i'll add a separate test for that 👍

File: src/select/__tests__/select.test.js (hunk 2 above)
Comment (schnerd) [reply src/select/__tests__/select.test.js]: backspace no longer changes internal component state, it emits an `onChange` event, and all of these test were just asserting on component state and not checking for calls to `onChange`. keeping the backspace test here would have resulted in a few more conditional statements that make this code more convoluted.

**Pull request merged**